src/Tokens.hpp

```diff
@@ -0,0 +1,398 @@
+//Anton Yarakhtin
+#pragma once
+#include <memory>
+#include <cmath>
+#include <unordered_map>
+#include <sstream>
+
+
+namespace Tokens 
+{
+	class Token;
+	using TokenMap = std::unordered_map<std::string, std::unique_ptr<Token>>;
+
+
+	enum class Type : int
+	{
+		NUMBER,
+		U_OPERATOR,
+		B_OPERATOR,
+		L_PARENTHESIS,
+		R_PARENTHESIS
+	};
+
+	const int operatorMaxLength = 4;
+
+	class Token {
+	public:
+		Token(Type type) { this->type = type; }
+		Type getType() const { return type; }
+		virtual std::string toString() { return std::string("#"); }
+		virtual int getPriority() const { return 0; }
+		virtual std::unique_ptr<Token> clone() { return nullptr; }
```

Comment:
This method should be pure virtual to avoid create default `Token`. Also this will force to create overload of `clone()` in inherited classes.